<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Телефонная книга</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:1, stop:0 #f5f7fa, stop:1 #c3cfe2);
    font-family: 'Segoe UI', Arial, sans-serif;
}

/* Общие стили для кнопок */
QPushButton {
    background-color: #4e73df;
    color: white;
    border: none;
    padding: 10px 20px;
    border-radius: 6px;
    font-size: 14px;
    font-weight: 500;
    min-width: 100px;
    text-align: center;
    transition: all 0.3s;
}

QPushButton:hover {
    background-color: #3a56c7;
    box-shadow: 0 4px 8px rgba(0,0,0,0.1);
}

QPushButton:pressed {
    background-color: #2e44a8;
}

/* Специфичные стили для кнопок */
QPushButton#deleteButton {
    background-color: #e74a3b;
}

QPushButton#deleteButton:hover {
    background-color: #d62c1a;
}

QPushButton#deleteButton:pressed {
    background-color: #ba2718;
}

QPushButton#sortButton {
    background-color: #36b9cc;
}

QPushButton#sortButton:hover {
    background-color: #2c9faf;
}

QPushButton#sortButton:pressed {
    background-color: #248391;
}

/* Стили для полей ввода */
QLineEdit, QTextEdit, QListWidget, QTableWidget {
    background-color: white;
    border: 1px solid #d1d3e2;
    border-radius: 4px;
    padding: 8px;
    font-size: 14px;
    selection-background-color: #4e73df;
    selection-color: white;
}

QLineEdit:focus, QTextEdit:focus {
    border: 1px solid #4e73df;
    outline: none;
}

/* Стили для QTabWidget */
QTabWidget::pane {
    border: 1px solid #d1d3e2;
    border-radius: 6px;
    margin: 5px;
    background: rgba(255,255,255,0.9);
}

QTabBar::tab {
    background: #f8f9fc;
    border: 1px solid #d1d3e2;
    border-bottom: none;
    border-top-left-radius: 6px;
    border-top-right-radius: 6px;
    padding: 8px 16px;
    margin-right: 2px;
    color: #5a5c69;
}

QTabBar::tab:selected {
    background: white;
    border-color: #d1d3e2;
    color: #4e73df;
    font-weight: bold;
}

QTabBar::tab:hover {
    background: #eaecf4;
}

/* Стили для таблицы */
QTableWidget {
    gridline-color: #e3e6f0;
    border-radius: 6px;
    alternate-background-color: #f8f9fc;
}

QHeaderView::section {
    background-color: #4e73df;
    color: white;
    padding: 8px;
    border: none;
    font-weight: 500;
}

QTableWidget QTableCornerButton::section {
    background-color: #4e73df;
    border: none;
}

/* Стили для списка */
QListWidget {
    border-radius: 6px;
}

QListWidget::item {
    padding: 10px;
    border-bottom: 1px solid #e3e6f0;
}

QListWidget::item:selected {
    background-color: #4e73df;
    color: white;
}

/* Стили для меток */
QLabel {
    color: #5a5c69;
    font-size: 14px;
    font-weight: 500;
    margin-bottom: 5px;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" spacing="0">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <widget class="QWidget" name="viewTab">
       <attribute name="title">
        <string>Просмотр контактов</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2" spacing="10">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout" spacing="10">
          <item>
           <widget class="QLineEdit" name="searchEdit">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>40</height>
             </size>
            </property>
            <property name="placeholderText">
             <string>Введите имя, телефон или адрес...</string>
            </property>
            <property name="styleSheet">
             <string notr="true">padding: 10px;
font-size: 14px;
border-radius: 6px;</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="sortButton">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string>Сортировать А-Я</string>
            </property>
            <property name="objectName">
             <string>sortButton</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTableWidget" name="contactsTable">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>1</verstretch>
           </sizepolicy>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="showGrid">
           <bool>true</bool>
          </property>
          <property name="gridStyle">
           <enum>Qt::DotLine</enum>
          </property>
          <property name="cornerButtonEnabled">
           <bool>true</bool>
          </property>
          <property name="columnCount">
           <number>4</number>
          </property>
          <property name="rowCount">
           <number>0</number>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <attribute name="horizontalHeaderItem0">
           <property name="text">
            <string>ФИО</string>
           </property>
          </attribute>
          <attribute name="horizontalHeaderItem1">
           <property name="text">
            <string>Телефон</string>
           </property>
          </attribute>
          <attribute name="horizontalHeaderItem2">
           <property name="text">
            <string>Email</string>
           </property>
          </attribute>
          <attribute name="horizontalHeaderItem3">
           <property name="text">
            <string>Адрес</string>
           </property>
          </attribute>
          <attribute name="horizontalHeaderVisible">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>200</number>
          </attribute>
          <attribute name="horizontalHeaderMinimumSectionSize">
           <number>50</number>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="editDeleteTab">
       <attribute name="title">
        <string>Удаление/Редактирование</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3" spacing="10">
        <item>
         <widget class="QListWidget" name="contactsList">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>1</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2" spacing="10">
          <item>
           <widget class="QPushButton" name="editButton">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string>Редактировать</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="deleteButton">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string>Удалить</string>
            </property>
            <property name="objectName">
             <string>deleteButton</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="addTab">
       <attribute name="title">
        <string>Добавить контакт</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4" spacing="10">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>ФИО:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="nameEdit">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>40</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">padding: 10px;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Телефон:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="phoneEdit">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>40</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">padding: 10px;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Email:</string>
           </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="emailEdit">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>40</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">padding: 10px;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Адрес:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="addressEdit">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>100</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">padding: 10px;</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="addButton">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>Добавить контакт</string>
          </property>
          <property name="styleSheet">
           <string notr="true">padding: 12px;
font-size: 15px;
font-weight: bold;</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>